src/code/z_demo.c

```diff
@@ -1,99 +1,1674 @@
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_BLAST) == ITEM_MASK_BLAST) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_SCENTS) == ITEM_MASK_SCENTS) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_GIANT) == ITEM_MASK_GIANT) {
+        count++;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EC678.s")
+    return count;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EC6D4.s")
+// Command 0xA: Textbox
+void Cutscene_Command_Textbox(GlobalContext* globalCtx, CutsceneContext* csCtx, CsCmdTextbox* cmd) {
+    static s32 D_801BB160 = CS_TEXTBOX_TYPE_DEFAULT;
+    u8 dialogState;
+    s32 pad;
+    u16 originalCsFrames;
+    s32 pad2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EC924.s")
+    if ((cmd->startFrame >= csCtx->frames) || ((cmd->endFrame < csCtx->frames))) {
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800ECD7C.s")
+    if (cmd->type != CS_TEXTBOX_TYPE_LEARN_SONG) {
+        if (D_801BB124 != cmd->base) {
+            if (D_801BB160 == CS_TEXTBOX_TYPE_3) {
+                csCtx->frames--;
+            }
+            D_801BB160 = CS_TEXTBOX_TYPE_1;
+            D_801BB124 = cmd->base;
+            if (cmd->type == CS_TEXTBOX_TYPE_BOSSES_REMAINS) {
+                if (CHECK_QUEST_ITEM(QUEST_REMAINS_ODOWLA) && CHECK_QUEST_ITEM(QUEST_REMAINS_GOHT) &&
+                    CHECK_QUEST_ITEM(QUEST_REMAINS_GYORG) && CHECK_QUEST_ITEM(QUEST_REMAINS_TWINMOLD)) {
+                    if (cmd->textId1 != 0xFFFF) {
+                        Message_StartTextbox(globalCtx, cmd->textId1, NULL);
+                    }
+                } else {
+                    Message_StartTextbox(globalCtx, cmd->base, NULL);
+                }
+            } else if (cmd->type == CS_TEXTBOX_TYPE_ALL_NORMAL_MASKS) {
+                if (Cutscene_CountNormalMasks() == 20) {
+                    if (cmd->textId1 != 0xFFFF) {
+                        Message_StartTextbox(globalCtx, cmd->textId1, NULL);
+                    }
+                } else {
+                    Message_StartTextbox(globalCtx, cmd->base, NULL);
+                }
+            } else {
+                Message_StartTextbox(globalCtx, cmd->base, NULL);
+            }
+        } else {
+            goto else_label;
+        }
+    } else if (D_801BB128 != cmd->base) {
+        D_801BB160 = CS_TEXTBOX_TYPE_LEARN_SONG;
+        D_801BB128 = cmd->base;
+        func_80152434(globalCtx, cmd->base);
+    } else {
+    else_label:
+        if (csCtx->frames >= cmd->endFrame) {
+            // The Textbox command can changee the current cutscene frame, mainly to prevent advancing the cutscene when
+            // a textbox that is expected to be closed by the user is still open.
 
+            originalCsFrames = csCtx->frames;
+            dialogState = Message_GetState(&globalCtx->msgCtx);
+            if (dialogState != 2 && dialogState != 0 && dialogState != 7 && dialogState != 8) {
```

Comment:
I'd rather you used brackets everywhere, especially since there are several places round here where you're doing so already.